### PR TITLE
Convert List to transient props

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?

Convert List to use transient props.

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

### CODE REVIEW

File: src/js/components/List/List.js [hunk 1]
```diff
@@ -679,9 +681,9 @@ const List = React.forwardRef(
                     tag="li"
```
Comment (jcfilben) [inline src/js/components/List/List.js:681]: Can we alphabetize

File: src/js/components/List/List.js [hunk 2]
```diff
@@ -679,9 +681,9 @@ const List = React.forwardRef(
...
                     background={adjustedBackground}
                     border={adjustedBorder}
-                    isDisabled={isDisabled}
                     flex={false}
                     pad={pad || theme.list.item.pad}
+                    $isDisabled={isDisabled}
                     {...defaultItemProps}
                     {...boxProps}
                     {...clickProps}
```
Comment (jcfilben) [inline src/js/components/List/List.js:689]: Do any of the props passed through here need to be transient?

Comment (taysea) [reply src/js/components/List/List.js:689]: Great question, I will take a look.

**Comment by jcfilben** [comment]
Closing in favor of styled components approach in https://github.com/grommet/grommet/pull/6947

**Pull request closed without merging**